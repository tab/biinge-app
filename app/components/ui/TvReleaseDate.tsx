import React from "react"

import { formatDate } from "helpers/formatDate"
import Typography from "components/ui/Typography"
import { tvStyles } from "styles"

type Props = {
  in_production: boolean
  release_date: string
  end_date: string
}

const TvReleaseDateComponent = ({
  in_production,
  release_date,
  end_date,
}: Props) => {
  return (
    <Typography variant="headline" style={tvStyles.date}>
      {in_production ? (
        <>{formatDate(release_date, "d MMMM yyyy")}</>
      ) : (
        <>
          {formatDate(release_date, "dd.MM.yyyy")}
          &nbsp;&ndash;&nbsp;
          {formatDate(end_date, "dd.MM.yyyy")}
        </>
      )}
    </Typography>
  )
}

export default TvReleaseDateComponent
