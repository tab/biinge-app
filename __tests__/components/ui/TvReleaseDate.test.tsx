import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"

import { it } from "@jest/globals"

import TvReleaseDate from "components/ui/TvReleaseDate"

describe("TvReleaseDate", () => {
  it("renders correctly when show is not ended", () => {
    const release_date = "2023-12-15"
    const end_date = ""

    const tree = render(
      <TvReleaseDate
        in_production={true}
        release_date={release_date}
        end_date={end_date}
      />,
    )
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly when show is ended", () => {
    const release_date = "2024-01-01"
    const end_date = "2024-02-27"

    const tree = render(
      <TvReleaseDate
        in_production={false}
        release_date={release_date}
        end_date={end_date}
      />,
    )
    expect(tree).toMatchSnapshot()
  })
})
