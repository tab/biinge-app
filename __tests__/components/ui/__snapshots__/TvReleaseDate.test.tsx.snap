// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvReleaseDate renders correctly when show is ended 1`] = `
<Text
  ellipsizeMode="tail"
  style={
    [
      {
        "color": "#181818",
        "fontSize": 17,
        "fontWeight": "600",
        "lineHeight": 22,
      },
      {
        "color": "#666666",
      },
    ]
  }
>
   – 
</Text>
`;

exports[`TvReleaseDate renders correctly when show is not ended 1`] = `
<Text
  ellipsizeMode="tail"
  style={
    [
      {
        "color": "#181818",
        "fontSize": 17,
        "fontWeight": "600",
        "lineHeight": 22,
      },
      {
        "color": "#666666",
      },
    ]
  }
>
   – 
</Text>
`;
